src/content/articles/kubernetes-for-scala-developers-part-one/index.mdx

```diff
@@ -0,0 +1,1382 @@
+We can also visualize the actual context using `kubectl config view`.
+
+To deploy resources in a specific namespace: run `kubectl -n development apply -f workload-api.yaml` where `n` is an
+argument taking as value the name of the Namespace.
+
+Response:
+```text
+secret/workload-secret created
+configmap/workload-config created
+deployment.apps/workload-api created
+service/workload-service created
+ingress.networking.k8s.io/api-workload-ingress created
+horizontalpodautoscaler.autoscaling/workload-horizontal-autoscaling created
+```
+To view the resources created in a specific namespace run: `kubeclt -n development -f workload-api.yaml`
+
+Response:
+```text
+kubectl -n testing get -f workload-api.yaml
+NAME                     TYPE     DATA   AGE
+secret/workload-secret   Opaque   2      12m
+
+NAME                        DATA   AGE
+configmap/workload-config   1      12m
+
+NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/workload-api   1/1     1            1           12m
+
+NAME                       TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+service/workload-service   ClusterIP   10.107.244.110   <none>        9002/TCP   12m
+
+NAME                                             CLASS   HOSTS                 ADDRESS        PORTS   AGE
+ingress.networking.k8s.io/api-workload-ingress   nginx   k8splayworkload.com   192.168.49.2   80      12m
+
+NAME                                                                  REFERENCE                 TARGETS                         MINPODS   MAXPODS   REPLICAS   AGE
+horizontalpodautoscaler.autoscaling/workload-horizontal-autoscaling   Deployment/workload-api   cpu: 67%/75%, memory: 46%/75%   1         4         1          12m
+```
+
+## Zero down-time deployment
+We have defined a zero-downtime deployment in the attribute `strategy`. to see how this works we can change the actual version
+of the app that is running, and we'll see that Kubernetes does not stop all the running container before starting new ones. Instead,
+it progressively changes old instances(`pfpw:20.20.31`) with new instances having the new version/release(`pfpw:20.20.32`). Doing this there will be no time when
+the service will not be available. To apply a new version first launch the previous deployment and then run this command:
+`kubectl set image -f workload-api.yml workload-api=aidocking/pfpw:20.20.32`. In a production environment, new releases are expected to be automatically
+deployed when conditions are met and this dynamic is driven by a tool like Jenkins x.
+
+## Conclusion: Your Kubernetes Journey Begins
+
+Kubernetes isn't just a tool, it's a mindset. For Scala developers, it represents a powerful way to transform how we
+think about deploying and managing applications. Start small, experiment, and gradually build your Kubernetes expertise.
+In this article we have discovered the basics components of a Kubernetes cluster. how to create, observe, update and delete them.
```

Comment:
- ~basics~ basic
- replace fullstop with :
- In the next part, we will explore...
- ~Automatic~ automatic

Reply:
Done